packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,155 @@
-// @ts-nocheck
-import get from 'lodash/get';
-import got from 'got';
-import Response from 'got';
+import * as z from 'zod';
+import got, { HTTPError } from 'got';
+import sortBy = require('lodash/sortBy');
 
-type EarthdataGetTokenResponse = Response<{
-  body: {
-    access_token?: string,
-    token_type?: string,
-    expiration_date?: string
-  }
-}>;
+const TokenSchema = z.object({
+  access_token: z.string(),
+  expiration_date: z.string(),
+});
 
-type EarthdataPostTokenResponse = Response<{
-  body: {
-    access_token?: string,
-    expiration_date?: string
-  }
-}>;
+const PostTokenSchema = z.object({
+  access_token: z.string(),
+  token_type: z.string(),
+  expiration_date: z.string(),
+});
 
-export interface EarthdataLoginParams {
-  username: string,
-  password: string,
-  edlEnv: string,
-}
+type Token = z.infer<typeof TokenSchema>;
 
-/**
- * A class to simplify requests for the Earthdata Login token used for CMR
- *
- * @typicalname earthdataLogin
- *
- * @example
- * const { EarthdataLogin } = require('@cumulus/cmr-client');
- *
- * const earthdataLogin = new EarthdataLogin({
- *  "username": "my-username",
- *  "password": "my-password",
- *  "edlEnv": "my-edl-environment"
- * });
- */
-export class EarthdataLogin {
-  username: string;
-  password: string;
-  edlEnv: string;
+const GetTokenResponseBody = z.array(TokenSchema);
 
-  /**
-  * The constructor for the EarthdataLogin class
-  *
-  * @param {string} params.username - Earthdata Login username, required parameter
-  * @param {string} params.password - Earthdata Login password, required parameter
-  * @param {string} params.edlEnv - the Earthdata Login environment (['PROD', 'OPS', 'SIT', 'UAT'])
-  *
-  * @example
-  *
-  * {
-  *  "username": "janedoe",
-  *  "password": "password",
-  *  "edlEnv": "UAT"
-  * }
-  */
+const PostTokenResponseBody = z.array(PostTokenSchema);
```

Comment:
How many elements will this array have? 0, 1, or more? If the answer is "always 1", then we should define this as `z.tuple([PostTokenSchema])`.